jepsen/src/jepsen/control.clj

```diff
@@ -270,23 +284,29 @@
 (defn session
   "Wraps clj-ssh-session in a wrapper for reconnection."
   [host]
-  (rc/open!
-    (rc/wrapper {:open    (if *dummy*
-                            (fn [] [:dummy host])
-                            (fn [] (clj-ssh-session host)))
-                 :name    [:control host]
-                 :close   (if *dummy*
-                            identity
-                            ssh/disconnect)
-                 :log?    true})))
+  (binding [*host* host]
```

Comment:
Okay, so... this part I'm a little bit less sure about. I need to take a closer look at this code because I'm not super familiar with it any more, but rc/wrapper is actually *reopenable*, so I'm not sure whether it's a good idea to catch exceptions only on the *first* call to rc/open!. Could subsequent reopenings also throw? Perhaps... it would be a good idea to put the try/catch *inside* the open function, right around `(clj-ssh-session host)`, and also to only bind *host* for the call to (debug-data) there, so that we can remember the host for subsequent open attempts.